### DISCUSSION TITLE
Play 2.9.3 - HTTPS issue

### DISCUSSION THREAD

**Discussion by h-ngo** [playframework/playframework#12713] [Q&A] [answered]
We are in the process of upgrading our Play from 2.8.19 to 2.9.3. We encountered an issue with HTTPS which we are not able to find any clue. (HTTP works fine)
Play Version

2.9.3
Operating System

Ubuntu
JDK

JDK 17
Scala 2.13.14
Library Dependencies

Akka 2.9.3
Akka-http 10.6.3
Expected Behavior

    Call the Play app using "curl -kv https://localhost:4443/"
    Received a response
    Continue receiving a response in subsequent calls

Actual Behavior

    Call the Play app using "curl -kv https://localhost:4443/"
    Received a response
    On the 2nd call, got below error

    Immediate connect fail for ::1: No route to host
    Trying 127.0.0.1:4443...
    Connected to localhost (127.0.0.1) port 4443
    Initializing NSS with certpath: sql:/etc/pki/nssdb
    CAfile: none
    CApath: none
    loaded libnssckbi.so
    ALPN: offers h2,http/1.1
    timed out before SSL handshake
    Closing connection
    curl: (28) timed out before SSL handshake

    On the 3rd call, got timeout error
    Immediate connect fail for ::1: No route to host

    Trying 127.0.0.1:4443...
    connect to 127.0.0.1 port 4443 failed: Connection refused
    Failed to connect to localhost port 4443 after 0 ms: Couldn't connect to server
    Closing connection
    curl: (7) Failed to connect to localhost port 4443 after 0 ms: Couldn't connect to server

HTTPS seems to die after the successful 1st call. There's no any error in the logs.

We have also tried to revert back Akka to 2.8.3 and Akka-http 10.5.2, which we have been using with Play 2.8.19 for past years, still the same behavior.

**Comment by mkurz** [comment, marked as answer]
Please, PLEASE, do not cross post, we all get that you have a problem but there is no need to open two posts with the exact same wording.
Closing this issue here in favor of #12712

**Reply by h-ngo** [reply]
sorry. it's our first time posting. We were not sure which one was actively being monitored.
